src/main.cpp

```diff
@@ -8,35 +8,80 @@
 
 int main(int argc, char *argv[]) {
+    argparser::ArgumentParser parser("yapgt (yet another photogrammetry tool)");
+
+    /* Calibration Arguments */
+    argparser::Argument a_loadCalibration("loadcalibration", "Filepath to load calibration from");
+    argparser::Argument a_saveCalibration("savecalibration", "Filepath to save calibration to");
+    argparser::Argument a_calibrationImages("calibrationImages", "Images for Calibration");
+    argparser::Argument a_calibrationCalibrateRow("calibrateRow", "corners row for Calibration");
+    argparser::Argument a_calibrationCalibrateColumn("calibrateColumn", "corners column for Calibration");
+
+    argparser::Argument a_matchImages("matchImages", "Images to use for matching");
+
+
+    parser.addArgument(&a_loadCalibration);
+    parser.addArgument(&a_saveCalibration);
+    parser.addArgument(&a_calibrationImages);
+    parser.addArgument(&a_calibrationCalibrateRow);
+    parser.addArgument(&a_calibrationCalibrateColumn);
+    parser.addArgument(&a_matchImages);
+
+    parser.parseArguments(argc, argv);
+
     Calibration calb = Calibration();
 
+    if (!(a_loadCalibration.isFound() || a_calibrationImages.isFound())) {
+        cout << "Neither " << a_loadCalibration.getName() << " nor " << a_loadCalibration.getName() << " was supplied" << endl;
+        return -1;
+    }
 
-    filesystem::path calibrationFile("./resources/calibration.xml");
-    filesystem::path path("./resources/chessboard_images");
+    if (a_loadCalibration.isFound()) {
+        calb.loadCalibration(filesystem::path(a_loadCalibration.getValue<string>()));
+    }
 
-    if (!filesystem::exists(calibrationFile)) {
+    if (a_calibrationImages.isFound()) {
+        filesystem::path path(a_calibrationImages.getValue<string>());
         vector<filesystem::path> filepaths;
         for (const auto& entry : filesystem::directory_iterator(path)) {
             filepaths.push_back(entry.path());
         }
 
         // number of inner corners per chessboard row and column
-        unsigned int cornersRow = 6;
-        unsigned int cornersColumn = 9;
+
+        if (!a_calibrationCalibrateRow.isFound()) {
+            cout << "Argument " << a_calibrationCalibrateRow.getName() << " is required for calibration" << endl;
+            return -1;
+        }
+
+        if (!a_calibrationCalibrateColumn.isFound()) {
+            cout << "Argument " << a_calibrationCalibrateColumn.getName() << " is required for calibration" << endl;
+            return -1;
+        }
+
+        unsigned int cornersRow = a_calibrationCalibrateRow.getValue<int>();
+        unsigned int cornersColumn = a_calibrationCalibrateColumn.getValue<int>();
 
         calb.calibrate(filepaths, cv::Size(cornersColumn, cornersRow));
-        calb.saveCalibration(calibrationFile);
-    } else {
-        calb.loadCalibration(calibrationFile);
     }
 
-    /* calb.loadCalibration("./resources/calibration.xml"); */
+    if (a_saveCalibration.isFound()) {
```

Comment:
Maybe check if the calibration is loaded from a file. If it is the same file saving isn't necessary.

Reply:
See above